test/graaflib/graph_test.cpp

```diff
@@ -38,6 +38,21 @@ TYPED_TEST(GraphTest, VertexCount) {
   ASSERT_EQ(graph.vertex_count(), 2);
   ASSERT_TRUE(graph.has_vertex(vertex_id_2));
   ASSERT_EQ(graph.get_vertex(vertex_id_2), 20);
+
+  // WHEN - THEN
+  constexpr int specific_id = 2;
+  const auto vertex_specific_id{graph.add_vertex(30, specific_id)};
+  ASSERT_EQ(graph.vertex_count(), 3);
+  ASSERT_TRUE(graph.has_vertex(specific_id));
+  ASSERT_EQ(graph.get_vertex(specific_id), 30);
+
+  // WHEN - THEN
+  constexpr int colliding_id = 2;
+  const auto vertex_id_4{graph.add_vertex(40)};
+  ASSERT_EQ(graph.vertex_count(), 4);
+  ASSERT_TRUE(graph.has_vertex(colliding_id + 1));
+  ASSERT_EQ(graph.get_vertex(colliding_id), 30);
+  ASSERT_EQ(graph.get_vertex(colliding_id + 1), 40);
```

Comment:
In this test section, aren't we also dependent on the implementation specific vertex ID logic since we are checking whether the graph has a vertex with ID `colliding_id + 1`. I remember the original reason for the new `add_vertex` methods was to not be dependent on this implementation detail.

Would it also make sense to get rid of this `WHEN - THEN` section then?

Reply:
> In this test section, aren't we also dependent on the implementation specific vertex ID logic since we are checking whether the graph has a vertex with ID `colliding_id + 1`. I remember the original reason for the new `add_vertex` methods was to not be dependent on this implementation detail.
> 
> Would it also make sense to get rid of this `WHEN - THEN` section then?

If I understand correctly, do you mean that this portion of the test might need to be changed down the line because we are relying specifically on how the id is calculated? In that case, I think it's fair to say that this section is too reliant on how `add_vertex` is implemented! I will remove in the next commit

Reply:
Exactly, if we change the underlying ID generation logic this test would break. 

I would say that the newly added method is already sufficiently covered so we can safely remove this part